src/sd.jl

```diff
@@ -94,6 +94,41 @@ triangular part of the matrix is constrained to belong to the
 """
 struct PSDCone end
 
+function build_constraint(
+    _error::Function,
+    f::AbstractMatrix{<:AbstractJuMPScalar},
+    s::MOI.GreaterThan,
+    extra::PSDCone,
+)
+    @assert iszero(s.lower)
+    return build_constraint(_error, f, extra)
+end
```

Comment:
Adding a new cone is just a case of writing these `build_constraint` functions.

Reply:
It might be possible (for vector inequalities) to use MOI sets and do it with one or two functions, right?
Might just a need a function: `_known_cone`

Reply:
We should try to make sure this leads to a consistent interface.
It seems we would want to allow `@constraint(model, x >= y, cone)` to be synonymous to `@constraint(model, x - y in cone)`.

Reply:
Yes, it's trivial to implement this syntax for any AbstractVectorSet. I just haven't done more widely to check it makes sense to do it for PSDCone first.

Reply:
I'm wondering if we could have a cone trait so that we don't need to have a new build_constraint for every set.
That would also allow doing `@constraint(model, p >= q, SOSCone()) ` with SumOfSquares

Reply:
You don't need a new `build_constraint`. You can just have 

```Julia
function build_constraint(
    _error::Function,
    f::AbstractMatrix{<:AbstractJuMPScalar},
    s::MOI.GreaterThan,
    extra::Union{MOI.AbstractVectorSet,JuMP.AbstractVectorSet},
)
    @assert iszero(s.lower)
    return build_constraint(_error, f, extra)
end
```

The `PSDCone` definition should probably be `struct PSDCone <: JuMP.AbstractVectorSet end`.

Reply:
AbstractVectorSet means the elements of this sets are vectors. This is not the case for PSDCone, they are matrices, nor for SOSCone, they are polynomials (hence the need for reshape, etc... to vectorize into an MOI vector set). Moreover, some AbstractVectorSet would not be cone so it wouldn't make sense to compare them with an inequality.

Reply:
It's also pretty easy to just add a `Union` for the sets in JuMP.

Reply:
Yes, we can start with that. But what about MOI sets ? Do we make a union of all MOI conic sets ? And update it after each release of MOI ?

Reply:
> so it wouldn't make sense to compare its elements with an inequality.

Okay. Sure, it doesn't make sense to do `X >= Y, MOI.Complments(2)`, but if the rule is just `X >= Y, Set()` is short-hand for `X - Y in Set()`, then it's up to the user to choose. We don't have to make the JuMP syntax mathematically meaningful.

Reply:
Regardless to this discussion, I think that's best left to a separate PR. This one is for deprecating `@SDconstraint` in favor of `@constraint(model, X >= Y, PSDCone())`